meeting-notes/2018/2018-12-12--gui-and-in-web-browsers-weekly.md

```diff
@@ -0,0 +1,175 @@
+-   **Notetaker:** @lidel
+-   **Attendees:**
+    -   @hacdias
+    -   @lidel
+    -   @fsdiogo
+    -   @pgte
+    -   @olizilla
+    -   @terichadbourne
+
+  
+
+
+**Moderator checklist**
+
+-   Ensure that there is a notetaker ;-)
+-   Start recording
+-   Call for additional agenda items
+
+  
+
+
+**Agenda**
+
+_General discussions, decisions, etc._
+
+-   Start with Team Updates and Demos
+-   lidel: we should make a plan around OKRs (finalizing Q4 & drafting Q1)
+-   olizilla: Desktop release plan
+-   olizilla: infra needs around share.ipfs.io
+
+  
+
+
+**Team Updates**
+
+**@lidel**
+-   Done:
+    -   Released new Companion Beta and Stable with fix for Chrome Beta:[v2.6.2.12270](https://github.com/ipfs-shipyard/ipfs-companion/releases/tag/v2.6.2.12270) and[v2.6.3](https://github.com/ipfs-shipyard/ipfs-companion/releases/tag/v2.6.3) 
+    -   Moved Issue with Weekly Sync to WG-agnostic repo:<https://github.com/ipfs/team-mgmt/issues/790> 
+    -   Going over my backglog of bugzilla and github notifications
+        -   Fun fact:[streams got enabled in Firefox Nightly](https://github.com/ipfs/js-ipfs-http-client/issues/848#issuecomment-445566344)!
+    -   window.ipfs.enable(opts) (aka Bulk Permission Prompt) - PR[ipfs-companion/pull/619](https://github.com/ipfs-shipyard/ipfs-companion/pull/619) 
+-   Blocked:
+    -   window.ipfs.enable build fails on Jenkins, probably due to tarser override, need to look into this
+-   Next:
+    -    [ipfs-companion/pull/619](https://github.com/ipfs-shipyard/ipfs-companion/pull/619) 
+        -   Finish remaining chores, get it reviewed and release to Beta
+        -   Extract remaining TODOs to dedicated issues
+    -   Q4 retro/Q1 OKRs
+
+olizilla
+
+-   Done:
+    -   [Rework the status page](https://github.com/ipfs-shipyard/ipfs-webui/pull/902) to remove the peer distribution graph… geolocating all peers via ipfs-geoip creates a bunch of work for the browser, and a[runaway resource usage issue in go-ipfs](https://github.com/ipfs/go-ipfs/issues/5613)
+    -   fix[ipfs-cluster website dev mode](https://github.com/ipfs/ipfs-cluster-website/pull/45) to unblock their updates
+    -   Introduced IPFS to new people at the redecentralize meetup in London<https://twitter.com/shevski/status/1072120698087321601> (met with matrix and scuttlebutt folks too)
+    -   Supporting a first time contributor on ipfs-desktop - https&#x3A;//github.com/ipfs-shipyard/ipfs-desktop/pull/726#issuecomment-445795049
+-   Blocked:
+    -   I was blocked by laryngitis. I’,m on the mend now.
+-   Next:
+    -   2 more usability testing session for webui. write up the common issues. Propose next round of improvements to webui
+    -   Refactoring to webui and ipfs-redux-bundle to work with the api changes in ipfs-http-client \*\*help wanted\*\*
+    -   Refactoring ipld-explorer-components to work with changes to dag-pb and ipld apis. \*\*help wanted\*\* (good opportunity to pair on ipld-explorer code.)
```

Comment:
I'll be available on Monday and Tuesday to help with this - updated WebUI blocks JS IPFS 0.34 release right now and I wanted it to happen this year.

Reply:
of note, Web UI is using older versions of js-ipld and js-ipfs-api (the lib formally known as). Right now, as long as I don't upgrade ipld, Web UI still works as intended, which is what I'd expect. I will need to upgrade the ipld explrorer UI when I take the new ipld lib, but I expect it to keep working until i do. 

Same goes for the file api from js-ipfs. I'd assume that older clients would still be able to access the http api in the same way, and new clients would use the new way. Or is this a breaking change to the http api too?

Reply:
There's no (intended) breaking change to the HTTP API.

To be honest, I used it on the [DHT branch](https://github.com/ipfs/js-ipfs/pull/856) and got a blank page for the files page...I didn't look into it any further at the time, so it just needs some investigation. I initially thought it was because of the breaking core API changes but as @olizilla pointed out the WebUI is bundled with it's dependencies, so _should_ still work (since there's no HTTP API breaks).